src/Bicep.Core/SourceLink/LinkedSourceFileKind.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Bicep.Core.SourceLink
+{
+    public enum LinkedSourceFileKind
+    {
+        Bicep,
```

Comment:
Please make sure we ignore source files in an archive's metadata file that have a file kind we don't recognize - for forwards compat.  Do we have a test that does that?

Reply:
Previously, we [threw an ArgumentException](https://github.com/Azure/bicep/blob/a2a85d0e930bee8352e05532ec77ebf1c27736c1/src/Bicep.Core/SourceLink/SourceArchive.cs#L240) for unrecognized file kinds. The updated implementation now throws a `JsonException`, which is consistent with existing behavior. Do we want to revisit this approach? But if we choose to ignore unknown file kinds, wouldn't the metadata version prevent that anyway - since a new file kind would likely come with a newer metadata version?